src/aleph_client/commands/instance/__init__.py

```diff
@@ -321,3 +322,126 @@ async def list(
             # Since we filtered on message type, we can safely cast as InstanceMessage.
             messages = cast(List[InstanceMessage], resp.messages)
             await _show_instances(messages)
+
+
+@app.command()
+async def expire(
+    vm_id: str,
+    domain: str,
+    private_key: Optional[str] = typer.Option(sdk_settings.PRIVATE_KEY_STRING, help=help_strings.PRIVATE_KEY),
+    private_key_file: Optional[Path] = typer.Option(sdk_settings.PRIVATE_KEY_FILE, help=help_strings.PRIVATE_KEY_FILE),
+    debug: bool = False,
+):
+    """expire an instance"""
+
+    setup_logging(debug)
+    account = _load_account(private_key, private_key_file)
+
+    async with VmClient(account, domain) as manager:
+        status, result = await manager.expire_instance(vm_id=vm_id)
+        if status != 200:
+            typer.echo(f"Status : {status}")
+        typer.echo(result)
+
+
+@app.command()
+async def erase(
+    vm_id: str,
+    domain: str,
+    private_key: Optional[str] = typer.Option(sdk_settings.PRIVATE_KEY_STRING, help=help_strings.PRIVATE_KEY),
+    private_key_file: Optional[Path] = typer.Option(sdk_settings.PRIVATE_KEY_FILE, help=help_strings.PRIVATE_KEY_FILE),
+    debug: bool = False,
+):
+    """erase an instance"""
+
+    setup_logging(debug)
+
+    account = _load_account(private_key, private_key_file)
+
+    async with VmClient(account, domain) as manager:
+        status, result = await manager.erase_instance(vm_id=vm_id)
+        if status != 200:
+            typer.echo(f"Status : {status}")
+        typer.echo(result)
+
+
+@app.command()
+async def reboot(
+    vm_id: str,
+    domain: str,
+    private_key: Optional[str] = typer.Option(sdk_settings.PRIVATE_KEY_STRING, help=help_strings.PRIVATE_KEY),
+    private_key_file: Optional[Path] = typer.Option(sdk_settings.PRIVATE_KEY_FILE, help=help_strings.PRIVATE_KEY_FILE),
+    debug: bool = False,
+):
+    """reboot an instance"""
+
+    setup_logging(debug)
+
+    account = _load_account(private_key, private_key_file)
+
+    async with VmClient(account, domain) as manager:
+        status, result = await manager.reboot_instance(vm_id=vm_id)
+        if status != 200:
+            typer.echo(f"Status : {status}")
+        typer.echo(result)
+
+
+@app.command()
+async def notify(
+    vm_id: str,
+    domain: str,
+    private_key: Optional[str] = typer.Option(sdk_settings.PRIVATE_KEY_STRING, help=help_strings.PRIVATE_KEY),
+    private_key_file: Optional[Path] = typer.Option(sdk_settings.PRIVATE_KEY_FILE, help=help_strings.PRIVATE_KEY_FILE),
+    debug: bool = False,
+):
+    """notify crn the instance, can be use to start VM"""
```

Comment:
The name notify don't make much sense, the actual endpoint is notify_allocation and it's the allocation part that is important. After discussion with Hugo we propose to change it.


I'm actually I'm wondering if it's really useful to present it to user, since it should be part of a higher command that take care of setting up the payment stream (and create the message) But maybe it's useful for debug?

```suggestion
@app.command()
async def allocate(
    vm_id: str,
    domain: str,
    private_key: Optional[str] = typer.Option(sdk_settings.PRIVATE_KEY_STRING, help=help_strings.PRIVATE_KEY),
    private_key_file: Optional[Path] = typer.Option(sdk_settings.PRIVATE_KEY_FILE, help=help_strings.PRIVATE_KEY_FILE),
    debug: bool = False,
):
    """Tell the CRN to start an instance with Pay as you go""
```

Reply:
I agree, this command should be added automatically on the PAYG instance creation workflow. But IMO I think that we can maintain this command for the user, because if a token stream gets empty of tokens, the CRN will stop the VM, and if the user re-activates the stream again with new tokens, he needs to be able to notify again the CRN to start again the VM. We also have the same feature on the frontend (in the next frontend release).